### PR TITLE
Dill locking & boosting UI

### PR DESCRIPTION
**Pull request by 0xkoffee**
Boosting integration needs to be finished after GaugeProxy contract change.

resolves pickle-finance/internal#1

### CODE REVIEW

**Review by mannyreimi88 — CHANGES_REQUESTED** [review]
@0xkoffee great job. Let's consider down-the-line to chunk the work and do the reviews on an ad-hoc basis if that's OK with you. From the reviewer's POV, it's hardcore to go through 22 files and 1,500+ lines of code in a single sitting.

A few matters:
- [x] no preview of how much `DILL` I am getting.
- [x] difficult to change dates without some buttons or presets. Consider adding "1 week", "1 month", "6 months", "1 year", "2 years", "3 years", and "4 years".
- [x] all "lock for" dates are expressed in weeks. For larger numbers, users may be more familiar with "months" or even "years" as units of measurement i.e. "4 years" rather than "208 weeks"
- [x] default lock for is 1 "weeks" instead of "week".
- [x] if there is a decay in the voting power of `DILL` this is not reflected in the UI.

The last matter I am not sure about, but if the user will need to authorize the app for spending `PICKLEs` there is no indication of this in the UI.

**Comment by mannyreimi88** [comment]
@larrythecucumber321 some preliminary observations on the `/gauges` route based on our discussions and this screenshot:



- [ ] give **Boost** its own element in a card-like format, like the **Active** box, similar to what we did with the **Zap** element on the home page.
- [ ] some instruction would be welcome, for example, **Input a weight for the boost you'd like to give each gauge. The app will automatically determine the final distribution of your boosts and confirm before applying them**.
- [ ] consider a two-step process:
    - [ ] first, the user inputs his desired weights on each gauge
    - [ ] on clicking **Boost**, a modal window appears, with the distribution (i.e. if the user added 1, 1, 1, to gauges a, b, and c he'll get a modal window saying gauge a 33.33%, gauge b 33.33%, gauge c 33.33%). Clicking **Apply Boosts** will set the transaction.
- [ ] the **Boost** card should only appear active for users who have locked DILL, otherwise, it should direct users to lock in DILL (if they have PICKLE, otherwise prompt them to acquire PICKLE).
- [ ] the current "Boost" (i.e. 1.000x) should appear in the card, and "Boost" to receive (i.e. 2.500x) should appear in the confirmation modal.

File: features/DILL/Claim.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    disabled: claimable ? false : true,
+    text: `Claim ${claimable ? formatNumber(claimable, 3) : "0"} PICKLEs`,
+  });
+  const {
+    status: transferStatus,
+    transfer,
+    getTransferStatus,
+  } = ERC20Transfer.useContainer();
+
+  const dillAPY =
```
Comment (Cipioo) [inline features/DILL/Claim.tsx:75]: This appears to be calculating the amount of return per week. multiply by 52 to get apy.

File: features/DILL/Claim.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    dillStats.weeklyDistribution && dillStats.lockedValue
+      ? dillStats.weeklyDistribution / dillStats.lockedValue
```
Comment (Cipioo) [inline features/DILL/Claim.tsx:77]: This appears to be calculating the amount of return per week. multiply by 52 to get apy. #
```suggestion
      ? dillStats.weeklyDistribution / dillStats.lockedValue * 52
```

Comment (Cipioo) [reply features/DILL/Claim.tsx:77]: If you want it more accurate you can multiply by 52.17857

**Review by larrythecucumber321 — APPROVED** [review]

**Pull request merged**